langchain4j-neo4j/src/main/java/dev/langchain4j/store/embedding/neo4j/Neo4jEmbeddingStore.java

```diff
@@ -0,0 +1,244 @@
+import static dev.langchain4j.internal.ValidationUtils.*;
+import static java.util.Collections.singletonList;
+
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.SessionConfig;
+
+import static dev.langchain4j.store.embedding.neo4j.Neo4jEmbeddingUtils.*;
+
+/**
+ * Represents a Vector index as an embedding store.
+ * Annotated with `@Getter` to be used in {@link Neo4jEmbeddingUtils}
+ */
+@Getter
+public class Neo4jEmbeddingStore implements EmbeddingStore<TextSegment> {
+
+    private static final Logger log = LoggerFactory.getLogger(Neo4jEmbeddingStore.class);
+
+    /* Neo4j Java Driver settings */
+    private final Driver driver;
+    private final SessionConfig config;
+
+    /* Neo4j schema field settings */
+    private final int dimension;
+    private final Neo4jDistanceType distanceType;
+
+    private final String indexName;
+    private final String metadataPrefix;
+    private final String embeddingProperty;
+    private final String label;
+    private final String text;
+
+    /**
+     * Creates an instance of Neo4jEmbeddingStore defining a {@link Driver} 
+     * starting from uri, user and password
+     */
+    public static class Neo4jEmbeddingStoreBuilder {
+        public Neo4jEmbeddingStoreBuilder withBasicAuth(String uri, String user, String password) {
+            return this.driver(GraphDatabase.driver(uri, AuthTokens.basic(user, password)));
+        }
+    }
+
+    /**
+     * Creates an instance of Neo4jEmbeddingStore
+     * @param driver: the {@link Driver} (required)
+     * @param dimension: the dimension (required)
+     * @param config: the {@link SessionConfig}  (optional, default is `SessionConfig.forDatabase("neo4j")`)
+     * @param label: the optional label name (default: "Document")
+     * @param property: the optional property name (default: "embedding")
+     * @param distanceType: the optional distanceType (default: "cosine")
+     * @param metadataPrefix: the optional metadata prefix (default: "")
+     * @param text: the optional text property name (default: "text")
+     * @param indexName: the optional index name (default: "langchain-embedding-index")
+     */
+    @Builder
+    public Neo4jEmbeddingStore(
+            SessionConfig config,
+            Driver driver,
+            int dimension,
+            String label,
+            String property,
+            Neo4jDistanceType distanceType,
+            String metadataPrefix,
+            String text,
+            String indexName) {
+        
+        /* required configs */
+        ensureNotNull(driver, "driver");
+        ensureNotNull(dimension, "dimension");
+
+        this.driver = driver;
+        this.dimension = dimension;
+        
+        /* optional configs */
+        this.config = getOrDefault(config, SessionConfig.forDatabase("neo4j"));
+        this.label = getOrDefault(label, DEFAULT_LABEL);
+        this.embeddingProperty = getOrDefault(property, DEFAULT_EMBEDDING_PROP);
```

Comment:
I'm missing the builder methods for all these other config options? are they somewhere else?

Reply:
The getter and builder method are auto-generate via[ lombok annotations.](https://github.com/langchain4j/langchain4j/pull/282/commits/36cd2f46c1280e53068bd2b9cd26941e17dfb0ac#diff-89b713731c6e14e29b6208dc8c14a92aff1a1baad94b44b8a9b388db8be97eb9R86)